UniversalAlgebra/Structures/Homs.lagda

```diff
@@ -0,0 +1,528 @@
+---
+layout: default
+title : Structures.Homs
+date : 2021-06-22
+author: [the ualib/agda-algebras development team][]
+---
+
+\begin{code}
+
+{-# OPTIONS --without-K --exact-split --safe #-} -- cubical #-}
+
+module Structures.Homs where
+
+open import Agda.Builtin.Equality  using    ( _≡_   ;  refl     )
+open import Agda.Primitive         using    (  _⊔_  ;  lsuc     )
+                                   renaming (  Set  to Type     )
+open import Data.Product           using    (  _,_  ;  Σ
+                                            ;  _×_  ;  Σ-syntax )
+                                   renaming ( proj₁ to fst
+                                            ; proj₂ to snd     )
+open import Level                  using    ( Level ;  Lift
+                                            ; lift  ;  lower    )
+                                   renaming ( zero  to ℓ₀       )
+open import Function.Base          using    ( _∘_   ;  id      )
+open import Relation.Binary.PropositionalEquality using    ( cong -- trans    ;   cong
+--                                                           ; cong-app
+                                                           ; module ≡-Reasoning )
+
+
+
+
+
+open import Overture.Preliminaries   using ( ∣_∣ ; ∥_∥ ; _∙_ ; _⁻¹)
+open import Overture.Inverses        using ( IsInjective ; IsSurjective )
+open import Relations.Discrete       using ( _|:_ ; 0[_] ; ker )
+open import Relations.Quotients      using ( Equivalence ; Quotient
+                                           ; 0[_]Equivalence ; ker-IsEquivalence ; kerlift-IsEquivalence
+                                           ; ⟪_⟫ ; ⌞_⌟ ; ⟪_∼_⟫-elim ; _/_ )
+open import Relations.Extensionality using ( swelldef )
+
+open import Structures.Basic         using ( Signature ; Structure ; _ʳ_ ; _ᵒ_ ; Compatible
+                                           ; Lift-Strucʳ ; Lift-Strucˡ ; Lift-Struc )
+open import Structures.Congruences
+
+
+private variable 𝑅 𝐹 : Signature
+
+
+-- Development for Structures (Sigma type representation)
+
+module _ {α ρ β ρ' : Level}
+         (𝑨 : Structure {α} {ρ} 𝑅 𝐹)
+         (𝑩 : Structure {β} {ρ'} 𝑅 𝐹) where
+
+ CompRel : (fst 𝑅) → ((fst 𝑨) → (fst 𝑩)) → Type (α ⊔ ρ ⊔ ρ')
```

Comment:
Same as with the previous - abbreviations would help readability a lot.

Reply:
And why is the naming convention different?

Reply:
I'm continuously going through and revising code, choosing better names, cleaning up stuff.  I guess I didn't get around to this one yet.